### DISCUSSION TITLE
keyball44:一部キーが反応しない

### DISCUSSION THREAD

**Discussion by Omanju23** [Yowkees/keyball#644] [Q&A]
お世話になっております。初めての自作キーボードで右も左もわからないため相談させてください。
[3-8．ProMicroの書き込みと動作確認]にて、各キーの動作確認をしたところ、SW23とSW1が動作しませんでした。（他のキーは動作しました。またSW23は２種類試しましたが、いずれもNGです。）

基盤図面も初めて見ているので見方があっているかわかりませんが、以下選択部分に接触不良があると判断し、はんだ付けを何度か試行してみましたが、状況変わらずでした…

#### SW23
![image](https://github.com/user-attachments/assets/7d978e36-82f3-4fb9-b5d4-7f4b6b4c9d0f)

#### SW1
![image](https://github.com/user-attachments/assets/7463da2d-8290-41bc-83b6-9543c5bee017)

以下、実際の基盤の写真です。

![基盤写真](https://github.com/user-attachments/assets/8f6ef1ce-aae1-44b8-8340-006ba2064e25)

怪しい点や、試行錯誤してみたほうがいい点などアドバイスいただけますと幸いです…！

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

不具合のあるキーが点在していますので、ご指摘通りダイオードもしくはキーソケットの半田付け不良と思われます。
囲って示していただいたダイオードで合っています。

それと、特に多いのがキーソケットのはんだ付け不良です。基板のパターンとキーソケット端子の隙間に半田が流れ込んでいるか、横から覗いて確認してみてください。

ピンセットでキーソケット左右をショートして、文字が入力されるかも確認いただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Reply by Omanju23** [reply]
お忙しいところご回答ありがとうございます。
回路図の見方があってることが分かっただけで、安心してリトライできるようになりました。
週末、ダイオードとキーソケットのはんだ付けをリトライしてみようと思います！
